### PR TITLE
Bob G's Release Notes for CNV 2.6

### PR DESCRIPTION
**Pull request by bgaydosrh**
Release notes in draft for 2.6:

See Netify link for Test Build.

Thanks

Bob

### CODE REVIEW

File: virt/virt-2-6-release-notes.adoc [hunk 1]
```diff
@@ -59,8 +59,9 @@ Other operating system templates shipped with {VirtProductName} are not supporte
 
 //CNV-9455: Improvements to KubeVirt and CDI ensure that storage is allocated on the correct node
 
-//CNV-9456: When importing VM disk images, you can now control which multus network is used
+* While a disk image is prepared for consumption by a virtual machine, it is possible that the virtual machine cannot be scheduled to the node where the local storage PV was previously pinned. In this release, `StorageClass` with `volumeBindingMode` is set to `WaitForFirstConsumer`, and the binding and provisioning of the PV is delayed until a `Pod` is created using the PVC.
```
Comment (aglitke) [inline virt/virt-2-6-release-notes.adoc:62]: For this one I'd focus more on the apparent impact to the user rather than the technical details.  For example:

When preparing local storage for a virtual machine disk image a volume may be allocated from a different node than is required by the VM.  To prevent scheduling problems, this release integrates with the kubernetes WaitForFirstConsumer binding mode to ensure that volumes are allocated from the correct node.

File: virt/virt-2-6-release-notes.adoc [hunk 2]
```diff
@@ -88,6 +89,16 @@ Other operating system templates shipped with {VirtProductName} are not supporte
 
 //CNV-9288: Blank block disks and VMware imported disk images are not preallocated even if requested
 
+* The Containerized Data Importer (CDI) and KubeVirt depend on QEMU which does not support NFS version 3. Therefore, only NFS version 4 is supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1892445[*BZ#1892445*])
+
+* The name of the Fedora PVC in the `openshift-virtualization-os-images` namespace is `fedora`, instead of `fedora32`. If you populated the `fedora32` PVC in {VirtProductname} 2.5 or earlier, the virtual machine does not appear in the web console and you cannot use it to clone another virtual machine. To workaround this issue, upload a Fedora image by naming the PVC `fedora`, instead of `fedora32`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1913352[*BZ#1913352*])
+
+* When creating a HPP boot source, the data volume is `pending` with a `WaitForFirstConsumer` status if a user creates the boot source using the `Import via URL (creates PVC)` option. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1929177[*BZ#1929177*])
```
Comment (aglitke) [inline virt/virt-2-6-release-notes.adoc:96]: This note implies that something is still broken but I see the bug is targeted to 2.6.0 and ON_QA.  Do we need to reword this to say that this issue has been fixed?  Also, I don't believe this issue has ever existed in a released version so why do we need to state that it is fixed?

File: virt/virt-2-6-release-notes.adoc (hunk 1 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:62]: Thanks Adam, that's much better. Appreciate the big picture view, I wasn't getting it. Moving to QE Review.

File: virt/virt-2-6-release-notes.adoc (hunk 2 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:96]: Thanks @aglitke  My bad for not checking the status of the bug. Yes, it makes no sense to document it. 

One question: Would this bug have impacted a z stream release? And if so, does that require we doc it? Just a thought.

But if you agree that a z would have no impact, then I will obsolete the story and update the BZ accordingly.

**Comment by duyanyan** [comment]
The note for known issue BZ#1913352 LGTM

File: virt/virt-2-6-release-notes.adoc [hunk 3]
```diff
@@ -91,6 +92,28 @@ Other operating system templates shipped with {VirtProductName} are not supporte
 
 //BZ-1855182 (https://bugzilla.redhat.com/show_bug.cgi?id=1855182): [Storage] Clone could not be continued after virtctl stop the vm if the clone dv have been created for more than 3 minutes
 
+* The Containerized Data Importer (CDI) and KubeVirt depend on QEMU which does not support NFS version 3. Therefore, only NFS version 4 is supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1892445[*BZ#1892445*])
+
+* The name of the Fedora PVC in the `openshift-virtualization-os-images` namespace is `fedora`, instead of `fedora32`. If you populated the `fedora32` PVC in {VirtProductname} 2.5 or earlier, the virtual machine does not appear in the web console and you cannot use it to clone another virtual machine. To workaround this issue, upload a Fedora image by naming the PVC `fedora`, instead of `fedora32`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1913352[*BZ#1913352*])
```
Comment (aburdenthehand) [inline virt/virt-2-6-release-notes.adoc:97]: Suggestion:
As a workaround, upload a Fedora image by naming the PVC `fedora` instead of `fedora32`.

File: virt/virt-2-6-release-notes.adoc [hunk 4]
```diff
@@ -91,6 +92,28 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
+
+* When creating a HPP boot source, the data volume is `pending` with a `WaitForFirstConsumer` status if a user creates the boot source using any method except the `Upload local file (creates PVC)` option. To resolve this issue, edit the YAML that defines the PVC in the Storage > PersistentVolumeClaims screen. Add the annotation `cdi.kubevirt.io/storage.bind.immediate.requested: "true"` to the YAML defining the persistent volume claim. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1929177[*BZ#1929177*])
```
Comment (aburdenthehand) [inline virt/virt-2-6-release-notes.adoc:99]: 1. Looking through the bug I gather 'Upload local file..' is GUI. in which case needs to be marked up with asteriks rather than backticks. Might also be useful to mention it is web console option. 
2. Storage > PersistentVolumeClaims should be `*Storage* -> *Persistent Volume Claims*` However this can also be done in the CLI so do we want to be GUI specific here?
3. If answer is 'no', suggestion for last two sentences:
`+`
`** As a workaround, edit the YAML of the underlying PVC of the data volume to add the cdi.kubevirt.io/storage.bind.immediate.requested: "true" annotation:`(<-- note, literal markup dropped for annotation only because it's messing with github markup. Annotation should still be wrapped in backticks)
`+`
`[source,yaml]`
`----`
`metadata:`
`  annotations:`
`    cdi.kubevirt.io/storage.bind.immediate.requested: "true"` (<-- note: looks like github dropping the necessary yaml spacing here and for annotations: line)
`----`
To make workaround more visible and include yaml example

Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:99]: Thanks Andrew, I appreciate that.

File: virt/virt-2-6-release-notes.adoc (hunk 3 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:97]: Done

File: virt/virt-2-6-release-notes.adoc (hunk 2 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc]: This is still a bug according to QE, though the scope is limited to HPP.

File: virt/virt-2-6-release-notes.adoc [hunk 5]
```diff
@@ -60,10 +60,12 @@ Other operating system templates shipped with {VirtProductName} are not supporte
 
 //CNV-9455: Improvements to KubeVirt and CDI ensure that storage is allocated on the correct node
 
-//CNV-9456: When importing VM disk images, you can now control which multus network is used
+* When preparing local storage for a virtual machine disk image, a volume may be allocated from a different node than is required by the virtual machine. To prevent scheduling problems, this release integrates with the Kubernetes `WaitForFirstConsumer` binding mode to ensure that volumes are allocated from the correct node.
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:63]: s/may/might

It is not clear what "this release" means. Suggestion: `s/this release/{VirtProductName} now` or `s/this release/the Containerized Data Importer (CDI) now` depending on your context.

File: virt/virt-2-6-release-notes.adoc [hunk 6]
```diff
@@ -95,6 +97,34 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
 
+* The Containerized Data Importer (CDI) and KubeVirt depend on QEMU which does not support NFS version 3. Therefore, only NFS version 4 is supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1892445[*BZ#1892445*])
+
+* The name of the Fedora PVC in the `openshift-virtualization-os-images` namespace is `fedora`, instead of `fedora32`. If you populated the `fedora32` PVC in {VirtProductname} 2.5 or earlier, the virtual machine does not appear in the web console and you cannot use it to clone another virtual machine. As a workaround, upload a Fedora image by naming the PVC `fedora`, instead of `fedora32`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1913352[*BZ#1913352*])
+
+* When creating a HPP boot source, the data volume is `pending` with a `WaitForFirstConsumer` status if a user creates the boot source using any method except the *Upload local file (creates PVC)* option. To resolve this issue, edit the YAML that defines the PVC in the *Storage* > *Persistent Volume Claims* screen.
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:104]: I don't think the second sentence "To resolve this..." is necessary since you talk about the workaround below.

File: virt/virt-2-6-release-notes.adoc [hunk 7]
```diff
@@ -95,6 +97,34 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
++
+** As a workaround, edit the YAML of the underlying PVC of the data volume to add the `cdi.kubevirt.io/storage.bind.immediate.requested: "true"` `annotation`:
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:106]: No need to mark this as a nested list item. Remove `**` at the start of the sentence.

File: virt/virt-2-6-release-notes.adoc [hunk 8]
```diff
@@ -95,6 +97,34 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
++
+[source,yaml]
+----
+metadata:
+annotations: cdi.kubevirt.io/storage.bind.immediate.requested: "true"
+----
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1929177[*BZ#1929177*])
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:113]: Suggestion to move this link at the end of the first sentence (assuming you remove the "To resolve this issue..." sentence).

File: virt/virt-2-6-release-notes.adoc [hunk 9]
```diff
@@ -95,6 +97,34 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
+
+* If you use a Fedora image as a boot source, it is no longer attached to a template if the PVC that you used to attach the boot source was previously provisioned. A new Fedora PVC with the name `fedora` should be attached to a template before you can use the template to create virtual machines from boot sources. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1907187[*BZ#1907187*]) (link:https://bugzilla.redhat.com/show_bug.cgi?id=1913352[*BZ#1913352*])
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:115]: Suggestion for the second sentence: "As a workaround, attach a new PVC with the name `fedora` to a template before using it to create virtual machines from boot sources."

File: virt/virt-2-6-release-notes.adoc [hunk 10]
```diff
@@ -95,6 +97,34 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
+
+//The issues below are from the 2.5 release. See items above for new known issues for 2.6.
+//The below issues have not yet been pruned/updated for 2.6.
+
+* When a node fails in user-provisioned installations of {product-title} on bare metal deployments, the virtual machine does not automatically restart on another node. Automatic restart is supported only for installer-provisioned installations that have machine health checks enabled. Learn more about xref:/../virt/install/preparing-cluster-for-virt#preparing-cluster-for-virt[configuring your cluster for {VirtProductName}].
+
+* If your {product-title} cluster uses OVN-Kubernetes as the default Container Network Interface (CNI) provider, you cannot attach a Linux bridge or bonding to the default interface of a host because of a change in the host network topology of OVN-Kubernetes. As a workaround, you can use a secondary network interface connected to your host or switch to the OpenShift SDN default CNI provider. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1887456[*BZ#1887456*])
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:122]: Not sure if these are your changes , but this known issue appears twice in the list. I am aware that the instance below references the correct BZ, so you can probably delete this.

**Review by sjhala-ccs — COMMENTED** [review]
A few suggestions, otherwise LGTM

File: virt/virt-2-6-release-notes.adoc (hunk 5 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:63]: Great, thanks.

File: virt/virt-2-6-release-notes.adoc (hunk 6 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:104]: Modified slightly and omitted sentence.

File: virt/virt-2-6-release-notes.adoc (hunk 7 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:106]: Ah, I wondered about this. Good, I don't like the ** either.

File: virt/virt-2-6-release-notes.adoc (hunk 8 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:113]: Done. Yeah, I forgot you usually list the bug before the workaround.

File: virt/virt-2-6-release-notes.adoc (hunk 9 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:115]: Done.

File: virt/virt-2-6-release-notes.adoc (hunk 10 above)
Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc]: Deleted 1887456

File: virt/virt-2-6-release-notes.adoc [hunk 11]
```diff
@@ -95,6 +97,31 @@ Other operating system templates shipped with {VirtProductName} are not supporte
...
+* The Containerized Data Importer (CDI) and KubeVirt depend on QEMU which does not support NFS version 3. Therefore, only NFS version 4 is supported. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1892445[*BZ#1892445*])
+
+* The name of the Fedora PVC in the `openshift-virtualization-os-images` namespace is `fedora`, instead of `fedora32`. If you populated the `fedora32` PVC in {VirtProductname} 2.5 or earlier, the virtual machine does not appear in the web console and you cannot use it to clone another virtual machine. As a workaround, upload a Fedora image by naming the PVC `fedora`, instead of `fedora32`. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1913352[*BZ#1913352*])
+
+* When creating a HPP boot source, the data volume is `pending` with a `WaitForFirstConsumer` status if a user creates the boot source using any method except the *Upload local file (creates PVC)* option. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1929177[*BZ#1929177*])
++
+As a workaround, in the *Storage* > *Persistent Volume Claims* web console screen, edit the YAML of the underlying PVC of the data volume to add the `cdi.kubevirt.io/storage.bind.immediate.requested: "true"` `annotation`:
```
Comment (sjhala-ccs) [inline virt/virt-2-6-release-notes.adoc:106]: Can you edit the YAML directly in the *Storage* > *Persistent Volume Claims* screen?

Comment (bgaydosrh) [reply virt/virt-2-6-release-notes.adoc:106]: Resolved this with Shikha. This feature is UI only for 2.6, just FYI

**Review by codyhoag — APPROVED** [review]
Approving release notes exception
